File: testdata/jennies/rawtypes/disjunctions_of_scalars_and_refs/PHPRawTypes/src/DisjunctionsOfScalarsAndRefs/MyRefA.php
<?php

namespace Grafana\Foundation\DisjunctionsOfScalarsAndRefs;

class MyRefA implements \JsonSerializable
{
    public string $foo;

    /**
     * @param string|null $foo
     */
    public function __construct(?string $foo = null)
    {
        $this->foo = $foo ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->foo !== $other->foo) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{foo?: string} $inputData */
        $data = $inputData;
        return new self(
            foo: $data["foo"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->foo = $this->foo;
        return $data;
    }
}

File: testdata/jennies/rawtypes/disjunctions_of_scalars_and_refs/JavaRawTypes/disjunctions_of_scalars_and_refs/MyRefA.java
package disjunctions_of_scalars_and_refs;

import java.util.Objects;

public class MyRefA {
    public String foo;
    public MyRefA() {
        this.foo = "";
    }
    public MyRefA(String foo) {
        this.foo = foo;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof MyRefA)) return false;
        MyRefA o = (MyRefA) other;
        if (!Objects.equals(this.foo, o.foo)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.foo);
    }
}
